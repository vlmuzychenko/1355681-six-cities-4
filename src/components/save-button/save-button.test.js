import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import NameSpace from "../../reducer/name-space.js";
import SaveButton from "./save-button.jsx";

const mockStore = configureStore([]);

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

it(`Should SaveButton render correctly`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
      authorizationData: {},
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <SaveButton
            isFavorite={true}
            authStatus={AuthorizationStatus.AUTH}
            offerId={1}
          />
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
