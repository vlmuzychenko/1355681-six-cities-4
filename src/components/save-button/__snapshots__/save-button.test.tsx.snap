// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should SaveButton render correctly 1`] = `
<button
  className="place-card__bookmark-button place-card__bookmark-button--active button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In
     bookmarks
  </span>
</button>
`;
